tests/unit/test_tracing.c

```diff
@@ -11,67 +11,91 @@
 
 SENTRY_TEST(basic_tracing_context)
 {
-    sentry_value_t span = sentry_value_new_object();
-    TEST_CHECK(sentry_value_is_null(sentry__span_get_trace_context(span)));
+    sentry_transaction_t *opaque_tx
+        = sentry__transaction_new(sentry_value_new_null());
+    TEST_CHECK(!opaque_tx);
 
-    sentry_value_set_by_key(span, "op", sentry_value_new_string("honk.beep"));
-    TEST_CHECK(sentry_value_is_null(sentry__span_get_trace_context(span)));
+    sentry_value_t tx = sentry_value_new_object();
+    opaque_tx = sentry__transaction_new(sentry__value_clone(tx));
+    sentry_value_set_by_key(tx, "op", sentry_value_new_string("honk.beep"));
+    TEST_CHECK(
+        sentry_value_is_null(sentry__transaction_get_trace_context(opaque_tx)));
 
     sentry_uuid_t trace_id = sentry_uuid_new_v4();
     sentry_value_set_by_key(
-        span, "trace_id", sentry__value_new_internal_uuid(&trace_id));
-    TEST_CHECK(sentry_value_is_null(sentry__span_get_trace_context(span)));
+        tx, "trace_id", sentry__value_new_internal_uuid(&trace_id));
+    sentry__transaction_decref(opaque_tx);
+    opaque_tx = sentry__transaction_new(sentry__value_clone(tx));
+    TEST_CHECK(
+        sentry_value_is_null(sentry__transaction_get_trace_context(opaque_tx)));
 
     sentry_uuid_t span_id = sentry_uuid_new_v4();
     sentry_value_set_by_key(
-        span, "span_id", sentry__value_new_span_uuid(&span_id));
+        tx, "span_id", sentry__value_new_span_uuid(&span_id));
+    sentry__transaction_decref(opaque_tx);
+    opaque_tx = sentry__transaction_new(sentry__value_clone(tx));
 
-    sentry_value_t trace_context = sentry__span_get_trace_context(span);
+    sentry_value_t trace_context
+        = sentry__transaction_get_trace_context(opaque_tx);
     TEST_CHECK(!sentry_value_is_null(trace_context));
     TEST_CHECK(!IS_NULL(trace_context, "trace_id"));
     TEST_CHECK(!IS_NULL(trace_context, "span_id"));
 
     CHECK_STRING_PROPERTY(trace_context, "op", "honk.beep");
 
     sentry_value_decref(trace_context);
-    sentry_value_decref(span);
+    sentry_value_decref(tx);
+    sentry__transaction_decref(opaque_tx);
 }
 
 SENTRY_TEST(basic_transaction)
 {
-    sentry_value_t tx_cxt = sentry_value_new_transaction_context(NULL, NULL);
+    sentry_transaction_context_t *opaque_tx_cxt
+        = sentry_transaction_context_new(NULL, NULL);
+    TEST_CHECK(opaque_tx_cxt != NULL);
+
+    sentry_value_t tx_cxt = opaque_tx_cxt->inner;
```

Comment:
looks like clang-analyzer does not like this:
```
[HIGH] /home/runner/work/sentry-native/sentry-native/tests/unit/test_tracing.c:57:29: Access to field 'inner' results in a dereference of a null pointer (loaded from variable 'opaque_tx_cxt') [core.NullDereference]
    sentry_value_t tx_cxt = opaque_tx_cxt->inner;
```

Its a bit annoying that its flagging test-only code. But sure enough. The assertion is not like in rust in the meaning that it panics and "returns" from the function. The function runs on and would hit a segfault here. In theory.